semantics/psc-trees-annotation.ads

```diff
@@ -54,6 +54,9 @@ package PSC.Trees.Annotation is
       New_Operand : Optional_Tree);
    --  Set Nth operand of given Tree
 
+   function Kind (T : Tree) return Tree_Kind_Enum;
```

Comment:
I suggest you replace these "Kind" declarations directly with expression functions, rather than postponing them to the body of their package.  There is no "information hiding" advantage to postponing these to the body, since they are always returning an enumeration literal that matches the type.